### PR TITLE
RIA-3996

### PR DESCRIPTION
**Pull request by Lee365**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3996

### Change description ###
RIA-3996 CCD reference visible to admin

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java [hunk 1]
```diff
@@ -69,6 +69,11 @@ public boolean canHandle(
 
         if (callback.getEvent() == Event.START_APPEAL) {
             asylumCase.write(APPEAL_REFERENCE_NUMBER, DRAFT);
+
+            String caseId = "" + callback.getCaseDetails().getId();
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java:73]: Would `String caseId = String.valueOf(callback.getCaseDetails().getId())` not work in this case. 

You may not need to run `replaceAll` on it afterwards.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/AsylumCaseFieldDefinition.java [hunk 2]
```diff
@@ -1400,6 +1400,8 @@
     DEPORTATION_ORDER_OPTIONS(
         "deportationOrderOptions", new TypeReference<YesOrNo>(){}),
 
+    CCD_REFERENCE_NUMBER_FOR_DISPLAY(
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/AsylumCaseFieldDefinition.java:1403]: I would consider calling the `CCD_CASE_ID` or `CCD_CASE_ID_FOR_DISPLAY` for better accuracy.

Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/AsylumCaseFieldDefinition.java:1403]: Left as CCD_REFERENCE_NUMBER_FOR_DISPLAY

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java (hunk 1 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java:73]: Changed to String caseId = String.valueOf(callback.getCaseDetails().getId());

**Review by bradsorour — DISMISSED** [review]
LGTM

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java [hunk 3]
```diff
@@ -69,6 +69,12 @@ public boolean canHandle(
 
         if (callback.getEvent() == Event.START_APPEAL) {
             asylumCase.write(APPEAL_REFERENCE_NUMBER, DRAFT);
+
+            String caseId = String.valueOf(callback.getCaseDetails().getId());
+            caseId = caseId.replaceAll("(.{" + 4 + "})", "$1 ").trim();
+
+            asylumCase.write(CCD_REFERENCE_NUMBER_FOR_DISPLAY,  caseId);
+
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java:77]: @Lee could you look to move this code into a new handler (i.e. a handler that will be triggered for all events) so the CCD_REFERENCE_NUMBER_FOR_DISPLAY is the available for all appeals.

If you take a look at the `AsylumCaseDataFixingHandler` class, you will see that the `canHandle()` method always returns true - so a handler similar to this. 

If we have the code in the `AppealReferenceNumberHandler`, the CCD_REFERENCE_NUMBER_FOR_DISPLAY will only display/be written to the AsylumCase for new cases.

**Comment by bradsorour** [comment]
@Lee365 a change to this PR if you can please (identified when planning NoC deployment with @makowalski)

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java (hunk 3 above)
Comment (makowalski) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java:77]: Yes, I agree. I think the cleanest way is to use new dedicated handler for it.
There is an option to use fixers but the whole part of the fixers should be refactored before they are more useful.

Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java]: Changed to use new handler - AsylumCaseCcdReferenceNumberHandler

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**